pythran/pythonic/numpy/partial_sum.hpp

```diff
@@ -13,24 +12,53 @@ namespace pythonic
   namespace numpy
   {
 
-    template <class Op>
-    template <class E, class F, class A>
-    A _partial_sum<Op, 1>::operator()(E e, F &o, A acc)
+    template <class Op, class A>
+    template <class E, class F>
+    A _partial_sum<Op, 1, A>::operator()(E e, F &o, A acc)
     {
       for (auto const &value : e) {
-        Op{}(acc, value);
+        acc = Op{}(acc, (A)value);
```

Comment:
This line look suspicious: why do you need the cast? It seem to prevent complex situation when accumulating over a numpy_expression.